consensus_encoding/src/decode/decoders.rs

```diff
@@ -102,6 +102,17 @@ impl Decoder for ByteVecDecoder {
     }
 }
 
+/// Implements `Decodable` for byte vectors.
+///
+/// This delegates to [`ByteVecDecoder`] which provides an optimized implementation
+/// that uses bulk copying rather than decoding bytes one at a time.
+#[cfg(feature = "alloc")]
+impl Decodable for Vec<u8> {
```

Comment:
This makes the types checkout nicely, but assumes any vec of bytes will have a compact size at the front. I believe this is an ok assumption in the bitcoin domain?